src/Datadog.Trace/Propagation/CommonHttpHeaderNames.cs

```diff
@@ -0,0 +1,14 @@
+namespace Datadog.Trace.Propagation
+{
+    /// <summary>
+    /// Names of HTTP headers that are commonly or internaly used
+    /// </summary>
+    public class CommonHttpHeaderNames
+    {
+        /// <summary>
+        /// If header is set to "false", tracing is disabled for that http request.
+        /// Tracing is enabled by default.
+        /// </summary>
+        public const string TracingEnabled = "otel-tracing-enabled";
```

Comment:
Should this be "x-otel-tracing-enabled" instead?

Reply:
1. @zacharycmontoya: ⚠️ this should be `x-datadog-tracing-enabled` for Datadog. make sure to change if this PR would be merged as it is

2. @pjanotti @nrcventura @zacharycmontoya : is it good that the "client" can disable tracing per a single request? I am not sure if having such a feature (especially by default) is good. Consider a bad actor who intentionally is disabled tracing to reduce the chances of being caught. Moreover, I have not found anything like that in https://github.com/open-telemetry/opentelemetry-specification 

I suggest reverting this change (to have `x-datadog-tracing-enabled` header) and create a dedicated GH issue to discuss the problem and address it separately.

Reply:
This seems to be dummy header anyway. In case if this header is in place all traces are dropped. More meant to control that exporters are not causing instrumentation loop.
I can see a problem when you are using internal / contrib integrations privately.

Reply:
This was a custom built solution by Datadog used primarily to avoid instrumenting the http payloads to our agent. I think we should follow the suggestion by @pellared and revert the value for now to `x-datadog-tracing-enabled` and we can figure out that story in a separate issue

Reply:
I agree with the recommendation from @zacharycmontoya. There are multiple ways to solve that problem, and we can consider them separately.